src/main/java/com/redhat/lightblue/camel/LightblueProducer.java

```diff
@@ -22,7 +24,20 @@ public LightblueProducer(LightblueScheduledPollEndpoint endpoint) {
     public void process(Exchange exchange) throws Exception {
         LightblueRequest req = (AbstractLightblueDataRequest) exchange.getIn().getBody();
 
-        exchange.getOut().setBody(endpoint.getLightblueClient().data(req));
+        try {
+            LightblueResponse response = endpoint.getLightblueClient().data(req);
+            if (response.hasError()) {
+                throw new RuntimeException("LightblueProducer:Error returned in response: " + response.getText());
```

Comment:
This is intentional, for the following reasons:
* client can throw LightblueException, and this pull request was to replace the verifier which threw LightblueErrorResponseException . 
* the intent is to wrap all producer exceptions ( including client exceptions that happens in the producer ) in a LightblueErrorResponseException , which can be handled in routes if needed ( as shown in tests )
* instead of setting exception on the exchange in two places, I decided to just throw an exception and setting it in the catch clause ( we will have to do that anyway)